File: AnnexNoSQL/src/Mongo/Prova3.java
package Mongo;

import org.bson.Document;

import com.mongodb.MongoClient;
import com.mongodb.client.FindIterable;
import com.mongodb.client.MongoDatabase;

public class Prova3 {

    public static void main(String[] args) {
        MongoClient con = new MongoClient("localhost" , 27017);
        MongoDatabase bd = con.getDatabase("test");
        
        Document ordenar = new Document();
        ordenar.put("precio", -1);
        
        FindIterable<Document> llibres =  bd.getCollection("libro").find().sort(ordenar);
        
        for (Document llibre : llibres)
            System.out.println("Títol:" + llibre.get("titulo") + ". Preu:" + llibre.get("precio"));
        
        con.close();
    }
}

File: Tema8_2/src/exemples/Prova3.kt
package exemples

import com.mongodb.MongoClient
import org.bson.Document
import java.util.logging.Level
import java.util.logging.LogManager

fun main(){
    LogManager.getLogManager().getLogger("").setLevel(Level.SEVERE)
    val con = MongoClient("localhost", 27017)
    val bd = con.getDatabase("test")

    val ordenar = Document()
    ordenar.put("precio", -1)

    val llibres = bd.getCollection("libro").find().sort(ordenar)

    for (llibre in llibres) System.out.println(
        "Títol: " + llibre.get("titulo").toString() + ". Preu: " + llibre.get("precio")
    )

    con.close()
}